test/powershell/Language/Scripting/ParameterBinding.Tests.ps1

```diff
@@ -0,0 +1,577 @@
+        },
+        @{
+            name = 'Parameter set';
+            cmd = "get-foo8 -b b a c d";   
+            expected=@('a', 'c', 'd')
+        },
+        @{
+            name = 'Default parameter set';
+            cmd = "get-foo9 a b c d";   
+            expected=@('a', $null, @('b','c','d'))
+        },
+        @{
+            name = 'Default parameter set 2';
+            cmd = "get-foo10 a b c d";   
+            expected=@($null, 'a', @('b','c','d'))
+        },
+        @{
+            name = 'Alias for parameters';
+            cmd = "get-foo11 -foo b";   
+            expected='b'
+        },
+        @{
+            name = 'Normal functions';
+            cmd = "foo12 a b";   
+            expected = 'b','a'
+        },
+        @{
+            name = 'AllowNull';
+            cmd = "foo14 -a `$null";   
+            expected = $null
+        },
+        @{
+            name = 'AllowEmptyString';
+            cmd = "get-foo16 -a ''";   
+            expected = ''
+        },
+        @{
+            name = 'AllowEmptyCollection';
+            cmd = "get-foo18 -a `@()";   
+            expected = $n
+        },
+        @{
+            name = "Unspecified non-mandatory bool does not cause exception";
+            cmd = "42 | get-foo19";
+            expected = 42
+        },
+        @{
+            name = "Parameter binding failure on Parameter1 does not cause parameter binding failure on Length"
+            cmd ="'abc' | get-foo20";
+            expected = 3
+        },
+        @{
+            name = "Binding array of string to array of bool succeeds"
+            cmd ="get-foo21a 'a','b'";
+            expected = @($true, $true)
+        },
+        @{
+            name = "no default value should not cause error"
+            cmd ="get-foo22c";
+            expected = $null
+        },
+        @{
+            name = "no default value should not cause error"
+            cmd ="get-foo22d";
+            expected = $null
+        },
+        @{
+            name = "Validation attributes should not run on default values"
+            cmd ="get-foo22e";
+            expected = 55
+        },
+        @{
+            name = "Validation attributes should not run on default values"
+            cmd ="get-foo22f";
+            expected = 55
+        },
+        @{
+            name = "Validation attributes should not run on default values"
+            cmd ="get-foo22g";
+            expected = $null
+        },
+        @{
+            name = "Validation attributes should not run on default values"
+            cmd ="get-foo22h";
+            expected = $null
+        },
+        @{
+            name = "Invoking with script block";
+            cmd = ". { param([Parameter(position=2)] `$a, [Parameter(position=1)]`$b); `$a; `$b} a b";
+            expected = @('b','a')
+        }        
+    )
+
+    It '<name>' -TestCases:$testData{
+        param($name, $cmd, $expected)
+        
+        $actual = Invoke-Expression $cmd
```

Comment:
fixed.

---
In reply to: [78435988](https://github.com/Powershell/Powershell/pull/2218#discussion_r78435988) [](ancestors = 78435988)